ansible_runner/config/_base.py

```diff
@@ -495,15 +494,17 @@ def wrap_args_for_containerization(self, args, execution_mode, cmdline_args):
 
         if self.container_auth_data:
             # Pull in the necessary registry auth info, if there is a container cred
-            host = self.container_auth_data.get('host')
-            username = self.container_auth_data.get('username')
-            password = self.container_auth_data.get('password')
-            self.registry_auth_path = self._generate_container_auth_file(host, username, password)
+            self.registry_auth_path, registry_auth_conf_file = self._generate_container_auth_dir(self.container_auth_data)
             if 'podman' in self.process_isolation_executable:
                 new_args.extend(["--authfile={}".format(self.registry_auth_path)])
             else:
                 docker_idx = new_args.index(self.process_isolation_executable)
                 new_args.insert(docker_idx + 1, "--config={}".format(self.registry_auth_path))
+            if registry_auth_conf_file is not None:
+                # Podman >= 3.1.0
+                self.env['CONTAINERS_REGISTRIES_CONF'] = registry_auth_conf_file
+                # Podman < 3.1.0
+                self.env['REGISTRIES_CONFIG_PATH'] = registry_auth_conf_file
```

Comment:
@shanemcd I am struggling with a decision here.

It would be more correct to set this in `os.environ` rather than `self.env`. It make no sense to expose this to the runtime in the container, but the file won't be viewable in the container.

I think `os.environ` would cause hesitations because it's expected that people use the python interface, and we have no clean way of reverting that. What we really need is to set it directly on the `pexepect_env` here:

https://github.com/ansible/ansible-runner/blob/cdc0961df51fa1e10b44371944aafe5ae140b98c/ansible_runner/runner.py#L178

But that would involve adding some other property on the RunnerConfig object, and there's already a lot. So because this isn't a regression, I'd prefer to be conservative and leave it like this (behaving the same it did before)

Reply:
I haven't had much time to let this thought marinade, so it might be dumb, but - since `container_auth_data` is a dictionary, it doesn't seem to crazy to shove another key under there and write the value to a file. Or something like that. WDYT?

Reply:
I don't really understand, I will try to catch up tomorrow.

Reply:
We spoke in Slack, what I proposed here is what they are already doing.